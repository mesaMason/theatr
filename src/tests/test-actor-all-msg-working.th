dolphin(int weight, int age):
    print("inside dolphin.")
    receive:
        swim():
            print("inside swim().")
        eat():
            print("inside eat().")
    drop:
        print("inside drop().")
    after:
        return;
        
func main() -> int:
    int weight = 10
    int age = 3
    actor d = new dolphin(weight, age)
    dolphin.asdfasdfsd() | d
    dolphin.eat() | d
    dolphin.swim() | d
    dolphin.die() | d    
    print("new dolphin created.")
    return 0

