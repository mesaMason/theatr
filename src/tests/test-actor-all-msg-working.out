new dolphin created.
inside dolphin.
inside drop().
inside eat().
inside swim().
